astropy/units/quantity.py

```diff
@@ -403,13 +405,20 @@ def __class_getitem__(cls, unit_shape_dtype):
         # TODO: ensure we do interact with NDArray.__class_getitem__.
         return Annotated.__class_getitem__((cls, unit))
 
-    def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
+    def __new__(cls, value, unit=None, dtype=np.floating, copy=True, order=None,
                 subok=False, ndmin=0):
 
         if unit is not None:
             # convert unit first, to avoid multiple string->unit conversions
             unit = Unit(unit)
 
+        # floating -> upcast to float dtype
```

Comment:
Maybe the intent is slightly clearer with:
```
float_default = dtype is np.floating
if float_default:
    dtype = None
```

Reply:
✅